session/bootstrap.go

```diff
@@ -1597,8 +1597,8 @@ func doDMLWorks(s Session) {
 		// Session only variable should not be inserted.
 		if v.Scope != variable.ScopeSession {
 			vVal := v.Value
-			if v.Name == variable.TiDBTxnMode && config.GetGlobalConfig().Store == "tikv" {
-				vVal = "pessimistic"
+			if v.Name == variable.TiDBTxnMode && config.GetGlobalConfig().Store != "tikv" {
+				vVal = "OPTIMISTIC"
```

Comment:
Maybe use `DefTiDBTxnMode` directly, is it possible?

Reply:
`DefTiDBTxnMode` is now pessimistic, but here we are saying if the store is not tikv use `OPTIMISTIC`. I can use `ast.Optimistic` if that helps?

Reply:
Oops, I did not notice that... No, just keep it unchanged. Do we need to change the document or so?

Reply:
Yes, the user documents should change. But I am hoping we can merge the automated docs generation and incorporate it here: https://github.com/pingcap/docs/pull/5720